docs/testing.md

```diff
@@ -0,0 +1,221 @@
+
+```js
+class GroceryShoppingListCreator extends React.Component {
+  render() {
+    return (
+      <>
+        <TextInput
+          placeholder="Enter grocery item"
+          onChangeText={this.setGroceryItem}
+        />
+        <Button
+          title="Add the item to list"
+          onPress={this.props.addNewItemToShoppingList}
+        />
+      </>
+    );
+  }
+}
+```
+
+When testing user interactions, test the component from the user perspective: you could write your tests to call `addNewItemToShoppingList` directly, and while such test works, it's very close to the implementation details of your component and would break if you refactor the component.
+
+To counter for that, component testing libraries such as [`react-native-testing-library`](https://github.com/callstack/react-native-testing-library), offer `fireEvent` apis that simulate a user interacting with the component. There are apis that allow to simulate entering text into textinput, tapping buttons and more. An example of how you may fire text change event using `react-native-testing-library`:
+
+```js
+test('updates grocery input', () => {
+  fireEvent.changeText(getByPlaceholder('Enter grocery item'), 'banana');
+});
+```
+
+The first parameter is a query function that returns a `ReactTestInstance` and the second parameter is the new text that user entered.
+
+This way, we're not testing what some function is doing - we're testing what happens when a user changes the text in the `TextInput`!
+
+Now that we know how to fire an event, let's take a look at verifying that it caused the expected change in your component: for example, that a certain text is rendered in a `GroceryShoppingList`. Aforementioned component testing libraries expose `query` apis for this purpose. An example may look like this:
+
+```js
+test('given empty GroceryShoppingList, user can add an item to it', () => {
+  const {getByTestId, getByText, getAllByText} = render(
+    <GroceryShoppingList />,
+  );
+
+  fireEvent.changeText(getByPlaceholder('Enter grocery item'), 'banana');
+  fireEvent.press(getByText('Add the item to list'));
+  const bananaElements = getAllByText('banana');
+  expect(bananaElements).toHaveLength(1); // expect 'banana' to be on the list
+});
+```
+
+### Testing Rendered Output
+
+[Snapshot testing](https://jestjs.io/docs/en/snapshot-testing) lets you verify how component is rendering. A snapshot is a textual representation of your component’s render output, and may look like this:
+
+```
+<Text
+  style={
+    Object {
+      "fontSize": 20,
+      "textAlign": "center",
+    }
+  }>
+  Welcome to React Native!
+</Text>
+```
+
+Snapshots are _generated_ by a component testing testing library, unlike the other types of tests which are written manually (snapshots are too complex to be created by hand).
```

Comment:
Let's remove this paragraph

Reply:
If you don't want to remove it, please at least remove the `testing testing` duplication, I forgot to mention earlier 😅

Reply:
@thymikee I did remove it in https://github.com/facebook/react-native-website/pull/1469/commits/4a04c6877333907f9e699af53915a68c0c6bc105